pkg/compression/compression.go

```diff
@@ -43,6 +49,7 @@ var compressionAlgos = map[string]struct {
 	"gzip":  {[]byte{0x1F, 0x8B, 0x08}, GzipDecompressor},                 // gzip (RFC 1952)
 	"bzip2": {[]byte{0x42, 0x5A, 0x68}, Bzip2Decompressor},                // bzip2 (decompress.c:BZ2_decompress)
 	"xz":    {[]byte{0xFD, 0x37, 0x7A, 0x58, 0x5A, 0x00}, XzDecompressor}, // xz (/usr/share/doc/xz/xz-file-format.txt)
+	"zstd":  {[]byte{0x25, 0xb5, 0x2f, 0xfd}, ZstdDecompressor},           // zstd (http://www.zstd.net)
```

Comment:
The current spec says
```c
#define ZSTD_MAGICNUMBER            0xFD2FB528    /* valid since v0.8.0 */
```
and the spec also says …8; apparently the last hexadecimal digit has historically been a protocol number, and the last update was 3 years ago.

What do we really want to do here? Using _only_ the 0.5 format seems rather surprising (although I know basically nothing about Zstd).

Reply:
good point, I've added only the version I've seen generated.  Should we list them all?  We probably want to add at last the current version (0.8.0)

@terrelln do you have any suggestions for us?

Reply:
I recommend only using the latest version. Versions before 0.8 were "prototype" formats. We stabilized zstd's format 3 years ago with version 1.0.0, and it will never change again.

Sadly, Ubuntu has zstd-0.5 in one of its older LTS releases, but we got them to update it to the latest zstd a year ago. I would start with only the latest zstd, and add 0.5 support if requested, which I think is unlikely.